src/compiler/checker.ts

```diff
@@ -9048,13 +9049,37 @@ export function createTypeChecker(host: TypeCheckerHost): TypeChecker {
         })!.parent;
     }
 
+    function getInstanceTypeOfClassSymbol(classSymbol: Symbol): Type {
+        const classType = getDeclaredTypeOfSymbol(classSymbol) as GenericType;
+        const objectFlags = getObjectFlags(classType);
+        if (!(objectFlags & ObjectFlags.ClassOrInterface) || !classType.typeParameters) {
+            return classType;
+        }
+        const variances = getVariances(classType);
+        const typeArguments = map(classType.typeParameters, (typeParameter, i) => {
+            if (!strictInstanceOfTypeParameters) {
+                return anyType;
+            }
+            const variance = variances[i];
+            switch (variance & VarianceFlags.VarianceMask) {
+                case VarianceFlags.Independent:
+                case VarianceFlags.Bivariant:
+                    return anyType;
+                case VarianceFlags.Invariant:
+                    return unknownType;
+            }
+            return getBaseConstraintOfType(typeParameter) || unknownType;
+        });
+        return createTypeReference(classType as GenericType, typeArguments);
+    }
+
     function getTypeOfPrototypeProperty(prototype: Symbol): Type {
         // TypeScript 1.0 spec (April 2014): 8.4
         // Every class automatically contains a static property member named 'prototype',
-        // the type of which is an instantiation of the class type with type Any supplied as a type argument for each type parameter.
+        // the type of which is an instantiation of the class type.
         // It is an error to explicitly declare a static property member with the name 'prototype'.

```

Comment:
Should I remove the "TypeScript 1.0 spec" line, since this has been edited and is no longer the same behavior?

Reply:
Sure, making the comments reflect the new state of the code is always a good idea.

Reply:
Okay cool, just wanted to make sure that I wasn't deleting some important historic artifact 😅